libs/wingsdk/src/core/internal.ts

```diff
@@ -103,8 +105,14 @@ function serializeImmutableData(obj: any): string {
         return `new Map(${serializeImmutableData(Array.from(obj))})`;
       }
 
-      // structs are just objects
-      if (obj instanceof Object) {
+      // if the object is a resource (i.e. has a "_toInflight" method"), then we lift and add to the
+      // list of bindings.
+      if (typeof (obj as IResource)._toInflight === "function") {
+        return (obj as IResource)._toInflight().text;
+      }
```

Comment:
This comment does not match this code.

BTW - I feel like it's okay to treat constructs without `_toInflight` or `_bind` as still resources - they're just resources that don't have any inflight functionality. For example we could provide a default `_toInflight()` implementation that returns an empty inflight client class, and a bind method that returns nothing.

Reply:
> BTW - I feel like it's okay to treat constructs without _toInflight or _bind as still resources - they're just resources that don't have any inflight functionality. For example we could provide a default _toInflight() implementation that returns an empty inflight client class, and a bind method that does nothing

That's an interesting idea. There is even a use case for referencing things like resource attributes from inflight.

```js
bring "aws-cdk" as aws;

let bucket = new aws.s3.Bucket();

let handler = inflight () => {
  print(bucket.bucket_name);
};
```

I'll open an issue and follow up.

Reply:
https://github.com/winglang/wing/issues/1251